.github/workflows/security.yml

```diff
@@ -0,0 +1,69 @@
+name: Security-check
+
+on:
+  pull_request: # change to push before merging code
+    branches:
+      - dev
+    paths:
+      - "**.py"
+      - ".github/workflows/*.yml"
+
+jobs:
+  Security-check:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      matrix:
+        python-version: [3.8]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Upgrade pip
+        run: |
+          pip install --upgrade --user pip
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+      - name: pip cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-py${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ matrix.python-version }}-
+      - name: Install dependencies
+        run: |
+          hash -r
+          pip install -r requirements.dev.txt
+          pip install -e .
+
+      - id: file_changes  # get changed files.
+        uses: trilom/file-changes-action@v1.2.3
+        with:
+          output: ' '
+
+      - uses: pre-commit/action@v2.0.0
+        with:
+          extra_args: --files ${{ steps.file_changes.outputs.files}}  # apply only to changed files.
+
+      - name: Check docstring coverage
+        run: |
+          interrogate -mvp src/ -f 80
+
+      - name: Scan for security issues
+        run: |
+          bandit -r src
+          safety check
+
+      - name: Synk security check
+        uses: snyk/actions/python-3.8@master
+        env:
+          SNYK_TOKEN: 3106d1f2-a045-4bc6-8695-d899ab653448 # to be changed to ${{ secrets.SNYK_TOKEN }}
```

Comment:
Q: What type of token is this ? Will it work ?

Reply:
So what it should look like is 
```
SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
```

But we need to set it up for the organization yet, so currently, the token you see is my personal token (to check if it works on CI). Once we see that it is working and someone with write access generates synk_token, then we can replace it with above snippet and it should work fine. Reference Link: https://github.com/snyk/actions#getting-your-snyk-token

Reply:
For the moment we depend on [this](https://github.com/OpenMined/PySyft/pull/5548) - after OpenMined has the token we can put it in.